nodePackages.typescript-language-server: add typescript dependency

kira-bruneau opened the pull request:
###### Motivation for this change
typescript-language-server is a wrapper around tsserver and is useless without it.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @malob

malob commented:
This looks good to me! (Works on my macOS machine.)

jD91mZM2 reviewed (approved):
I stumbled upon this after wondering why typescript-language-server didn't work like it should. Adding `typescript` as an installed package worked too - although this is obviously a much better solution.

prusnak commented:
Please rework your PR. It now has a merge conflict after PR https://github.com/NixOS/nixpkgs/pull/89184 has been merged

kira-bruneau committed:
nodePackages.typescript-language-server: add typescript dependency

typescript-language-server is a wrapper around tsserver and is useless
without it

MetaDark commented:
@prusnak I just force pushed a change to bring it up-to-date.

felschr commented:
Just wondering would a local tsserver installation still override the default one with these changes?

MetaDark commented:
@felschr No, right now it wouldn't. But I could update `PATH` using `--sufix` instead of `--prefix` so it would.

SuperSandro2000 commented:
PR got combined with a few others other node updates. Closing so that the update goes trough smoothly.
